staging/src/k8s.io/cloud-provider/config/types.go

```diff
@@ -100,4 +101,12 @@ type WebhookConfiguration struct {
 	// '-foo' means "disable 'foo'"
 	// first item for a particular name wins
 	Webhooks []string
+	// WebhookPort is the port that the controller-manager's webhook service runs on.
+	WebhookPort int32
+	// WebhookAddress is the IP address to serve on.
+	WebhookAddress string
+	// CaBundle is the ca certs to be used by apiserver for validating the webhook server certs.
+	CaBundle string
+	// ValidationWebhookConfiguration is the config used for creating validation webhook config object.
+	ValidationWebhookConfiguration *admissionregistrationv1.ValidatingWebhookConfiguration
```

Comment:
Do we want to allow the user to choose between validating and mutating webhooks on a per-webhook basis?

Reply:
yeah i think its possible to do it by taking another input here https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/app/webhooks.go#L56-L59